### ISSUE TITLE
no reconnect possible when using `zmq4.NewPub` with `socket.Dial`

### ISSUE THREAD

**Issue by thielepaul** [go-zeromq/zmq4#124]
I have the following code:
```go
socket := zmq4.NewPub(ctx)
socket.Dial(endpoint)
err := socket.SendMulti(zmq4.NewMsgFrom(partA, partB))
```

It works fine at first, but when I restart the other side of the connection at `endpoint` then there is neither an automatic reconnect nor can I do it manually, because the send method will not return an error.

My current workaround is to use reflection to check in the unexported `conns` field, if a connection is there, but I think there should be a better way to do this:
```go
func isConnected(socket zmq4.Socket) bool {
	return reflect.ValueOf(socket).Elem().FieldByName("sck").Elem().FieldByName("conns").Len() != 0
}
```

I could think of the following solutions:
- handle the reconnect internally
- add an `IsConnected() bool` method to the `Socket` interface so the user can reconnect on connection loss
- return an error from the `SendMulti` and `Send` methods to let the user know that connection is broken

**Comment by sbinet** [comment]
a) feels a bit too magical (do you know what does `libzmq++` ?)
b) is simple enough but might encourage racy code
c) might break some code but I guess that code might already be broken anyway.

baring a clear lead from the original C++ library, I'd vote for c).
would you volunteer providing a PR?

**Comment by thielepaul** [comment]
As far as I know `libzmq` does it internally. I could do a PR once we decide for one of the options

**Comment by thielepaul** [comment]
@sbinet what so think about that suggestion? #127

**Comment by thielepaul** [comment]
fixed in #127

@sbinet could you please tag a new release with the new features :)
